src/SonarScanner.MSBuild.PreProcessor/SonarWebService.cs

```diff
@@ -207,14 +192,9 @@ private static SonarRule CreateRule(JObject r, JToken actives)
 
             var projectId = GetProjectIdentifier(projectKey, projectBranch);
 
-            if (await IsSonarCloud() || (await GetServerVersion()).CompareTo(new Version(6, 3)) >= 0)
-            {
-                return await GetComponentProperties(projectId);
-            }
-            else
-            {
-                return await GetComponentPropertiesLegacy(projectId);
-            }
+            return await IsSonarCloud() || (await GetServerVersion()).CompareTo(new Version(6, 3)) >= 0
+                       ? await GetComponentProperties(projectId)
+                       : await GetComponentPropertiesLegacy(projectId);
         }
 
         public async Task<Version> GetServerVersion()
```

Comment:
This `GetServerVersion` and `DownlaodServerVersion` concept is weird. The relation is based on side effects and it's confusing that `DownloadServerVersion` doesn't return the version.

Let's just remove the `DownloadServerVersion`. It's called only here. And it will make this easy to understand